src/main/java/com/desiato/puresynth/controllers/LoginController.java

```diff
@@ -0,0 +1,31 @@
+package com.desiato.puresynth.controllers;
+
+import com.desiato.puresynth.dtos.*;
+import com.desiato.puresynth.exceptions.InvalidTokenException;
+import com.desiato.puresynth.services.AuthenticationService;
+import com.desiato.puresynth.services.UserService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/login")
+public class LoginController {
+
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    public LoginController(UserService userService, AuthenticationService authenticationService) {
+        this.userService = userService;
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping
+    public ResponseEntity<LoginResponseDTO> authenticateUser(@RequestBody AuthenticationRequestDTO requestDTO) throws InvalidTokenException {
```

Comment:
Before I showed you 2 versions to solve the problem with typing:
* external exception handler
* inheritance of response DTO

Implementing both of them doesn't make sense. You're always returning `SuccessLoginResponseDTO` from this method unless an exception is thrown. You don't need the inheritance structure of `LoginResponseDTO` when you have an exception handler.

Let's do this simplification:
* leave the exception handler
* don't use the interface `LoginResponseDTO` in successful or error responses. 
* rename `SuccessLoginResponseDTO` to just `LoginResponseDTO`
* in the exception handler you can still return `ErrorLoginResponseDTO`

Reply:
done!